### PR TITLE
Fixed #29462 -- Fixed ogrinspect test failures with GDAL 2.2.

### PR DESCRIPTION
**Pull request by timgraham**
In the version that gives `srid=-1`, `layer.srs` in https://github.com/django/django/blob/5008d59a2a8b9df09737fc05c9e343146a348a90/django/contrib/gis/utils/ogrinspect.py#L218-L227
is
```
GEOGCS["GCS_WGS_1984",
    DATUM["WGS_1984",
        SPHEROID["WGS_84",6378137,298.257223563]],
    PRIMEM["Greenwich",0],
    UNIT["Degree",0.017453292519943295]]
```
In the new version, it's:
```
GEOGCS["GCS_WGS_1984",
    DATUM["WGS_1984",
        SPHEROID["WGS_84",6378137,298.257223563]],
    PRIMEM["Greenwich",0],
    UNIT["Degree",0.017453292519943295],
    AUTHORITY["EPSG","4326"]]
```
I guess we might check GIS library release notes for something relevant.

### CODE REVIEW

File: tests/gis_tests/geos_tests/test_io.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def test_empty_polygon_wkb(self):
             wkb_w.byteorder = byteorder
             for srid, hex in enumerate(hexes):
                 wkb_w.srid = srid
-                self.assertEqual(wkb_w.write_hex(p), hex)
+                self.assertEqual(wkb_w.write_hex(p), hex)  # This line fails with b'0000000000' != b'000000000300000000'
```
Comment (timgraham) [inline tests/gis_tests/geos_tests/test_io.py]: @sir-sigurd, can you investigate or suggest what might be the issue?

File: tests/gis_tests/inspectapp/tests.py [hunk 2]
```diff
@@ -133,12 +133,12 @@ def test_time_field(self):
         ))
 
         # The ordering of model fields might vary depending on several factors (version of GDAL, etc.)
-        self.assertIn('    f_decimal = models.DecimalField(max_digits=0, decimal_places=0)', model_def)
+        self.assertIn('    f_decimal = models.CharField(max_length=0)', model_def)
```
Comment (timgraham) [inline tests/gis_tests/inspectapp/tests.py]: @claudep, do you think this issue could be something similar to ffdf507ec0821f0520e315c2e8a6cf231ea3fd5a but for float/double?

File: tests/gis_tests/inspectapp/tests.py [hunk 3]
```diff
@@ -75,15 +75,15 @@ def test_poly(self):
             '    float = models.FloatField()',
             '    int = models.{}()'.format('BigIntegerField' if GDAL_VERSION >= (2, 0) else 'FloatField'),
             '    str = models.CharField(max_length=80)',
-            '    geom = models.PolygonField(srid=-1)',
+            '    geom = models.PolygonField()',
```
[the remark this answers is not in the document]
Comment (timgraham) [reply tests/gis_tests/inspectapp/tests.py]: In the version that gives `srid=-1`, `layer.srs` in https://github.com/django/django/blob/5008d59a2a8b9df09737fc05c9e343146a348a90/django/contrib/gis/utils/ogrinspect.py#L218-L227
is
```
GEOGCS["GCS_WGS_1984",
    DATUM["WGS_1984",
        SPHEROID["WGS_84",6378137,298.257223563]],
    PRIMEM["Greenwich",0],
    UNIT["Degree",0.017453292519943295]]
```
In the new version, it's:
```
GEOGCS["GCS_WGS_1984",
    DATUM["WGS_1984",
        SPHEROID["WGS_84",6378137,298.257223563]],
    PRIMEM["Greenwich",0],
    UNIT["Degree",0.017453292519943295],
    AUTHORITY["EPSG","4326"]]
```
I guess we might check GIS library release notes for something relevant?

File: tests/gis_tests/geos_tests/test_io.py (hunk 1 above)
Comment (timgraham) [reply tests/gis_tests/geos_tests/test_io.py]: This looks like the change https://trac.osgeo.org/geos/ticket/680 (authored by Sergey himself). The change went in GEOS 3.6.1. I guess the test's expectations should be conditional on `geos_version_tuple()`.

Comment (sir-sigurd) [reply tests/gis_tests/geos_tests/test_io.py]: I can't check it properly as I'm on vacation, but I guess conditionals should be added to WKBWriter class. See https://github.com/django/django/commit/b90d72facf1e4294df1c2e6b51b26f6879bf2992#diff-181a3ea304dfaf57f1e1d680b32d2b76R226

Comment (timgraham) [reply tests/gis_tests/geos_tests/test_io.py]: Thanks for the pointer! Done in https://github.com/django/django/pull/10014.

**Comment by timgraham** [comment]
buildbot, test on bionic.

**Comment by timgraham** [comment]
buildbot, test on bionic.

**Pull request merged**